app/scripts/check_requirements

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env php
+<?php
+
+// Script should not be called from the Web
+if (php_sapi_name() != 'cli') {
+    die('Nope');
+}
+
+/**
+ * Colorize the shell output
+ *
+ * @param string $text The text.
+ * @param string $color Color of the text: green, red or nothing.
+ * @return string
+*/
+function colorizeOutput($text, $color)
+    {
+        switch ($color) {
+            case 'green':
+                // White on green
+                $output = "\033[1;37m\033[42m{$text}";
+                break;
+            case 'red':
+                // White on red
+                $output = "\033[1;37m\033[41m{$text}";
+                break;
+            default:
+                $output = $text;
+                break;
+        }
+        $output .= "\033[0m";
+        return $output;
+    }
+
+/**
+ * Check current configuration.
+ * 
+ * @param bool   $condition The type of alert.
+ * @param string $message   The message to show.
+ * @param string $help      The hint about the message.
+ * @param string $suggested_fix A message that help for fix the issue.
+ 
+ * @return void
+ */
+function check($condition, $message, $help = '', $suggested_fix = '')
+{
+    $text = "* {$message}: ";
+    if($condition) {
+        $text .= colorizeOutput($condition ? 'OK' : 'FAILED', 'green');
+    } else {
+        $text .= colorizeOutput($condition ? 'OK' : 'FAILED', 'red');
+    }
+    $text .= "\n";
+
+    if (! $condition) {
+        if(! empty($help)) { 
+            $text .=  "  {$help}\n";
+        }
+        if(! empty($suggested_fix)) {
+            $text .=  "  {$suggested_fix}\n";
+        }
+    }
+    
+    echo $text;
+}
+
+/**
+ * Check if the directory is empty
+ *
+ * @param string $dir The path.
+ * @return bool
+*/
+function isEmptyDir($dir)
+{
+    if (($files = @scandir($dir)) && count($files) <= 2) {
+        return true;
+    }
+        
+    return false;
+}
+
+echo " Transvision Requirements Check\n";
+
+// Get the parent folder from the script itself
+$root = dirname(dirname(dirname(__FILE__)));
+
+//Load settings and constants
+$server_config = parse_ini_file($root . '/app/config/config.ini');
+include_once($root . '/app/inc/constants.php');
+
+// Check php version
+$php_version = phpversion();
+check(version_compare($php_version, '5.4', '>='), sprintf('PHP version is at least 5.4 (%s)', $php_version), 'Current version is ' . $php_version);
+
+// Check permissions on cache folder for the user
+clearstatcache(null, CACHE_PATH);
+$cachePerms = substr(sprintf('%o', fileperms(CACHE_PATH) ), -4);
+$command = 'chmod -R 755 ' . CACHE_PATH;
+check($cachePerms[1] == '7', 'Cache folder is writable', sprintf('Cache folder should be writable (current permissions : (%s))', $cachePerms), "To fix execute: " . $command );
```

Comment:
Still extra space before the colon.

Note: the cache config could be fine even if this fails, i.e. the user running PHP is able to write in it. This should probably be a warning, saying: this is currently not writable, make sure it's writable by the user running webserver and PHP.